proto/messages.proto

```diff
@@ -28,6 +28,16 @@ message GenericMessage {
   }
 }
 
+message QualifiedUser {
+  optional string id = 1;
+  optional string domain = 2;
+}
+
+message QualifiedConversation {
+  optional string id = 1;
+  optional string domain = 2;
+}
```

Comment:
Why are the fields optional? `QualifiedUser` is already optional.

Reply:
I was going for maximum forward compatibility. In the docs it says 
> Some engineers at Google have come to the conclusion that using required does more harm than good; they prefer to use only optional and repeated. However, this view is not universal.

Not sure how necessary this is. Any preference on what should be optional?

Reply:
Ah, okay. I was just asking because we didn't use optional that much before.

> Any preference on what should be optional?

That depends on the spec :)